### ISSUE TITLE
NavigationActions.back() action is unexpected behavior

### ISSUE THREAD

**Issue by Gennady77** [react-navigation/react-navigation#3294]
Hi everyone.

I started to learn react navigation and saw that back action behavior is not like in a browser and unexpected for me. I created small app with the follow structure:

TabNavigator has 'Home', 'Notifications' and 'Topics' tabs.
'Topics' is a StackNavigator and has 'Topic1', 'Topic2' and 'Topic3' screens
'Topic1' has a button to navigate to 'Topic2'
'Topic2' has a button to navigate to 'Topic3'
(important) Tab 'Notification' has a button to navigate to 'Topic3'

see link https://snack.expo.io/r1DT5e6NM

I'm doing the navigation according to the following scheme:

Home -> Notifications -> Topic3 -> press 'back'

Expected result:

I should be returned from Topic3 screen to the Notification

Real result:

I navigate to 'Topic2' screen and when I press 'back' again I navigate to 'Topic1'

It's really unexpected behavior for me. So I need to understand this problem. Is it notmal for react navigation? Any ideas?

**Comment by kelset** [comment]
Closing because you haven't followed the issue template; please open a new one that follows the template.
Without that, it's nearly impossible to understand & investigate the issue. I suggest you also try (in the new issue) to answer/explain in a way that fits these questions:

1. have you tried updating to the latest release?
1. does it happen only on Android/iOS?
1. does it happen on both Debug and Release modes?
1. does it happen with Debugger On or Off? Both?
1. Can you provide a GIF or a repo or a snack to reproduce this issue?

That said, there have been many issues related to the goBack feature, so I suggest you check [those tagged that way](https://github.com/react-navigation/react-navigation/issues?utf8=%E2%9C%93&q=is%3Aissue+back+label%3A%22related%3A+goBack%22+) before opening a new issue. Chances are that this is already there.

**Comment by Gennady77** [comment]
I see none of these points in other opened issues but only my issue was closed. I included a link on working application. What else is needed?

**Comment by Gennady77** [comment]
You closed my issue again. What's the problem?

**Comment by kelset** [comment]
I am seriously baffled by your behaviour

> What's the problem?

1. You haven't respected the issue template
1. You haven't provided enough informations
1. You haven't even checked the issues tagged I linked in my first comment
1. I've already wrote these things in my **first comment** that you **completely overlooked**
1. You are not even reporting a bug, but simply an "unexpected behavior for you"
1. You haven't even read the README

Moreover you seem to think that issues are customer support tickets, so let me be **extremely clear here**: you're entitled to the level of support you pay for in open source and that's **none**.

So, start behaving in a **appropriate way** or I will ban you.
